dds/DCPS/InfoRepoDiscovery/InfoRepoDiscovery.cpp

```diff
@@ -619,9 +619,22 @@ InfoRepoDiscovery::add_publication(DDS::DomainId_t domainId,
     //this is the client reference to the DataWriterRemoteImpl
     OpenDDS::DCPS::DataWriterRemote_var dr_remote_obj =
       servant_to_remote_reference(writer_remote_impl, orb_);
+    //turn into a octet seq to pass through generated files
+    const DCPS::Encoding& encoding = XTypes::get_typeobject_encoding();
+    DCPS::Message_Block_Ptr data(new ACE_Message_Block(
```

Comment:
similar to the note above, this could be refactored to eliminate duplication with `add_type_info_param` in ParameterListConverter.cpp